src/test/java/serialization/ChsDeltaDeserializerTest.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.companieshouse.pscstatement.delta.serialization;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.delta.ChsDelta;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.pscstatement.delta.exception.NonRetryableErrorException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+@ExtendWith(MockitoExtension.class)
+public class ChsDeltaDeserializerTest {
+
+    @Mock
+    private Logger logger;
+    private ChsDeltaDeserializer deserializer;
+    @BeforeEach
+    public void init() {
+        deserializer = new ChsDeltaDeserializer(logger);
+    }
+
+    @Test
+    void When_deserialize_Expect_ValidChsDeltaObject() {
+        ChsDelta chsDelta = new ChsDelta("data", 1, "context_id");
+        byte[] data = encodedData(chsDelta);
+
+        ChsDelta deserializedObject = deserializer.deserialize("", data);
+
+        assertThat(deserializedObject).isEqualTo(chsDelta);
+    }
+
+    @Test
+    void When_deserializeFails_throwsNonRetryableError() {
+        byte[] data = "Invalid message".getBytes();
+        assertThrows(NonRetryableErrorException.class, () -> deserializer.deserialize("", data));
+    }
+
+    private byte[] encodedData(ChsDelta chsDelta){
+        ChsDeltaSerializer serializer = new ChsDeltaSerializer(this.logger);
```

Comment:
I'm not a fan of using a serializer to create a test payload here. It assumes the serializer serializes properly. Could we actually use the bytes from a real payload coming in from the chs-delta-api instead and set it up that way?

Reply:
Yeah that's probably better, I can also just delete the serializer for now then as it's not in the scope of this Jira